DMTN-306.tex

```diff
@@ -1,79 +1,119 @@
+
+\subsection{Registration of files to replicate}
+\label{registration}
+
+To perform replication, we create Rucio Datasets, each composed of a set of files that are already in their appropriate locations at the source RSE. Upon registration, pre-configured Rucio subscriptions trigger the actual file movement to the destination facility, in accordance with the defined replication rules. Rucio delegates the data transfer to FTS, which then instructs the storage endpoints at the facilities to move the data, typically by requesting the destination facility to pull the data from the source facility.
+
+Rubin has developed the tool \texttt{rucio\_register}\footnote{https://github.com/lsst/rucio\_register}, which allows for the selection of existing files from a Butler repository based on specified criteria. The tool attaches Rubin-specific metadata to these files and registers them into one or more Rucio Datasets. The metadata, encoded as a JSON record, contains a minimal set of information extracted from the origin Butler repository. This ensures that replicated files can be ingested properly into the local Butler repository at the destination facility.
+
+Only files that require replication to another facility are registered with Rucio. As a result, Rubin's instance of Rucio is aware only of files replicated across processing facilities. Files that are local to each facility and not subject to replication remain known only to that facility's Butler and are not registered in Rucio.
+
+\subsection{Ingestion at reception}
+\label{ingestion}
+
+FTS notifies Rucio about the completion of individual file transfers. Rubin's \texttt{HermesK}\footnote{https://github.com/lsst-dm/ctrl\_rucio\_ingest}, which is a modification of Rucio's Hermes daemon, filters messages and uses Kafka as a mechanism to signal the destination Rubin facility that a new file was replicated and to take appropriate actions.
+
+Messages distributed through Rubin's Kafka control-plane include Rubin-specific metadata. Those messages are received by Rubin's \texttt{ingestd}\footnote{https://github.com/lsst-dm/ctrl\_ingestd}, a daemon running at each destination facility responsible for ingesting newly replicated files into the local Butler repository.
 
-% ADD CONTENT HERE
-% You can also use the \input command to include several content files.
+Each facility only receives notifications about files successfully replicated to the storage endpoints it operates. This is achieved by following a simple convention: the name of Kafka topic the notification is sent to is named after the name of the Rucio storage element. Each facility's \texttt{ingestd} is configured to only monitor Kafka messages specifically targeted to the facility's RSEs (see figure \ref{fig:kafka-control-plane}).
```

Comment:
Maybe add:

In a complex distributed system such as this, having stateless daemons such as ingestd, idempotent transactions such as ingestion of file batches, and triggering off known synchronization points such as replication acknowledgement helps ensure a consistent, if conservative, view of the available data across multiple sites.

Reply:
Added